src/main/java/com/vmware/vipclient/i18n/messages/api/opt/local/LocalLocaleOpt.java

```diff
@@ -37,72 +39,104 @@ public LocalLocaleOpt(LocaleDTO dto) {
 	}
 
 	@Override
-    public Map<String, String> getSupportedLanguages(String locale) {
+    public void getSupportedLanguages(String locale, LocaleCacheItem cacheItem) {
 		Map<String, String> supportedLanguageNames = new HashMap<String, String>();
 		List<String> supportedLanguages = DataSourceEnum.Bundle.createProductOpt(dto)
```

Comment:
First get from service, then get from local. In this way, local L2 doesn't depend on local L3.

Reply:
This is in offline mode, so supportedLanguages local L2 need should get from local L3 bundle.